api/maven-api-core/src/main/java/org/apache/maven/api/Project.java

```diff
@@ -133,4 +134,10 @@ default String getId() {
 
     @Nonnull
     List<RemoteRepository> getRemotePluginRepositories();
+
+    /**
+     * Returns the effective project properties.
+     */
+    @Nonnull
+    Map<String, String> getProperties();
```

Comment:
I don't think introducing a mutable aspect here is a good idea.  This map should be made immutable imho.
Modifying project properties can be done using the `ProjectManager`.

Reply:
Ah, it's already immutable.  It should be written in the javadoc.  And maybe point to `ProjectManager#setProperty` ?